### PR TITLE
Avoid producing a few useless lambdas during conversion to DAML-LF

### PR DESCRIPTION
**Pull request by hurryabit**
This also removes a bit of code duplication by factoring out a common
pattern.

There are far more places where this technique can be applied. I'll do so
in a follow up PR.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/866)

### CODE REVIEW

File: daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs [hunk 1]
```diff
@@ -891,6 +877,20 @@ convertExpr env0 e = do
     convertArg env = \case
         Type t -> TyArg <$> convertType env t
         e -> TmArg <$> convertExpr env e
+    withTyArg :: Env -> (LF.TypeVarName, LF.Kind) -> [LArg Var] -> (LF.Type -> [LArg Var] -> ConvertM (LF.Expr, [LArg Var])) -> ConvertM (LF.Expr, [LArg Var])
```
Comment (neil-da) [inline daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs]: I'd rather these functions had a newline before them, and I think I'd even prefer if they were at the top level. They seem generic combinators with general properties.

**Comment by hurryabit** [comment]
My impression also is that the continuation can be simplified a bit, I haven't yet figured out how though. I'll land this in master and have another look afterwards.

**Pull request merged**